[FEATURE] [MER-938] Improve admin sections view

mcass19 opened the pull request:
[MER-938](https://eliterate.atlassian.net/browse/MER-938)

Improve the admin sections index so they are able to find them easier (specially if there a lot).
 
Changes include:
- Added some columns
- Sort by all columns
- Search by some more columns
- Filter by status, type or active by date

<img width="1439" alt="image" src="https://user-images.githubusercontent.com/42446726/164288660-477a297c-b8d2-4145-a058-cafd3f36b639.png">

Any feedback/suggestion/thought is welcome 😄 

**NOTE: This PR will be marked as "Ready for Review" and can be merged once it has 2 approvals.**

jrissler reviewed (approved):
This looks great and increases usefulness a great deal. Do you have any performance concerns?

Also does it look / work ok on mobile?

Otherwise great job :)

mcass19 commented:
In 69cc677712fa567346fdebe00308e3c6ec8e3b0b:
- Added some indexes to improve joins and filtering
- Digging a little bit more on how to perform full text searches, ended using `ts` based on [`postgreSQL docs`](https://www.postgresql.org/docs/current/textsearch-tables.html)
  - Added trgm indexes for searchable fields
  
@jrissler what do you think?

gastonabella reviewed (commented):
Great job! 👏

mcass19 commented on lib/oli_web/live/common/paged_table.ex:
```diff
@@ -12,23 +12,24 @@ defmodule OliWeb.Common.PagedTable do
+        {/if}
+
+        {#if @total_count > 0}
+          <Paging id="header_paging" total_count={@total_count} offset={@offset} limit={@limit} click={@page_change}/>
```
I think I didn't understand you. If results are less than `limit` we will have only one page.

mcass19 replied:
Yeah went in that direction to simplify and be consistent with the `Listing` component which is used in some other places and do the same. But maybe it's better to not show it when we have only one page (and we can update in both)

mcass19 commented on lib/oli_web/live/sections/sections_view.ex:
```diff
@@ -120,28 +106,73 @@ defmodule OliWeb.Sections.SectionsView do
+          <TextSearch id="text-search" text={@options.text_search}/>
+
+          <:extra_opts>
+            <Check checked={@options.active_date} click="active_date">Only active by date</Check>
```
haha agree 😄

gastonabella reviewed (approved):
LGTM

jrissler commented:
> In [69cc677](https://github.com/Simon-Initiative/oli-torus/commit/69cc677712fa567346fdebe00308e3c6ec8e3b0b):
> 
> * Added some indexes to improve joins and filtering
> * Digging a little bit more on how to perform full text searches, ended using `ts` based on [postgreSQL docs](https://www.postgresql.org/docs/current/textsearch-tables.html)
>   
>   * Added trgm indexes for searchable fields
> 
> @jrissler what do you think?

Looks good to me - that should help :) Great job
